uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/form/components/leave-type-wizard-form.component.js

```diff
@@ -12,7 +14,165 @@ define([], function () {
+      sharedSettings.sourcePath + 'leave-type-wizard/form/components/form-sections';
+    vm.settingsTabs = [
+      {
+        name: 'basic-details',
+        label: 'Basic',
+        fields: [
+          {
+            name: 'hide_label',
+            label: 'Hide leave type label on public calendars and feeds?',
+            helpText: [
+              'The CiviHR self service portal has an all staff calendar.',
+              'You can also create calendar feeds that can be integrated with your calendar app.',
+              lineBreak,
+              'If you enable this option the leave type will simply be given the title "Leave".',
+              'Managers and administrators will continue to see the actual leave title of the request.',
+              lineBreak,
+              'This may be helpful if the leave type title is sensitive - i.e. Unpaid leave or Compassionate leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'leave-requests',
+        label: 'Leave Requests',
+        fields: [
+          {
+            name: 'max_consecutive_leave_days',
+            label: 'Max consecutive duration (Leave blank for unlimited)',
+            helpText: [
+              'Configure the maximum duration of consecutive leave permitted to be selected in a single leave request.',
+              'You can leave this field blank for unlimited duration of leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'public-holidays',
+        label: 'Public Holidays',
+        fields: [
+          {
+            name: 'must_take_public_holiday_as_leave',
+            label: 'Do staff work on public holidays?'
+          }
+        ]
+      },
+      {
+        name: 'carry-forwards',
+        label: 'Carry Forwards',
+        fields: [
+          {
+            name: 'allow_carry_forward',
+            label: 'Allow carry forward?'
+          }
+        ]
+      }
+    ];
+
+    vm.$onInit = $onInit;
+    vm.getSettingsTabFields = getSettingsTabFields;
+    vm.openSection = openSection;
+    vm.openSettingsTab = openSettingsTab;
+    vm.selectLeaveTypeCategory = selectLeaveTypeCategory;
+
+    function $onInit () {
+      initDefaultView();
+    }
+
+    /**
+     * Gets fields for the currently active settings tab
+     *
+     * @return {Array} collection of fields
+     */
+    function getSettingsTabFields () {
+      var activeTab = _.find(vm.settingsTabs, { active: true });
+
+      return activeTab.fields;
+    }
+
+    /**
+     * Initiates the default view:
+     * - selects Leave category;
+     * - expands the General section and leaves the Settings section collapsed;
+     * - selects Basic Details settings tab.
+     */
+    function initDefaultView () {
+      vm.leaveTypeCategory = 'leave';
+
+      openSection('general');
+      openSettingsTab('basic-details');
+    }
+
+    /**
+     * Opens a section by its name and collapses all other sections
```

Comment:
but i thought that using Bootstrap's `.panel-group` was exactly for that? Are we reimplementing something that bootstrap already gives us?

https://github.com/civicrm/org.civicrm.shoreditch/blob/master/base/scss/vendor/bootstrap/_panels.scss#L219-L220

Also, remember we have access to angular's ui-bootstrap: http://angular-ui.github.io/bootstrap/#!#accordion

Reply:
I did consider this, but UI-bootstrap does not at least provide "do not collapse if opened" feature as far as I know.

Reply:
No, I was talking about collapsing and already expanded section, but yeah, `is-disabled` may be the way to go. But the thing it - the amount of code would still be the same, you will still need this `ng-class={ active: whatever }`